internal/provider/data_source_packer_image_iteration.go

```diff
@@ -0,0 +1,229 @@
+							Type:        schema.TypeList,
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"created_at": {
+										Description: "Creation time of this image.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"id": {
+										Description: "HCP ID of this image.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"image_id": {
+										Description: "Cloud Image ID, URL string identifying this image for the builder that built it.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+									"region": {
+										Description: "Region this image was built from. If any.",
+										Type:        schema.TypeString,
+										Computed:    true,
+									},
+								},
+							},
+						},
+						"labels": {
+							Description: "Labels for this build.",
+							Type:        schema.TypeMap,
+							Computed:    true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"packer_run_uuid": {
+							Description: "UUID of this build.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"status": {
+							Description: "Status this build. DONE means that all images tied to this build were successfully built.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"updated_at": {
+							Description: "Time this build was last updated.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePackerImageRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := d.Get("bucket").(string)
+	channelSlug := d.Get("channel").(string)
+	client := meta.(*clients.Client)
+
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	if err := setLocationData(d, loc); err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Reading HCP Packer registry (%s) [project_id=%s, organization_id=%s, channel=%s]", bucketName, loc.ProjectID, loc.OrganizationID, channelSlug)
+
+	channel, err := clients.GetPackerChannelBySlug(ctx, client, loc, bucketName, channelSlug)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	iteration := channel.Pointer.Iteration
+
+	d.SetId(channel.Pointer.Iteration.ID)
+
+	if err := d.Set("incremental_version", iteration.IncrementalVersion); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("created_at", iteration.CreatedAt.String()); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("builds", flattenPackerBuildList(iteration.Builds)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func setLocationData(d *schema.ResourceData, loc *sharedmodels.HashicorpCloudLocationLocation) error {
```

Comment:
Ooo, could make this a shared helper 👀

Reply:
I'll maybe do that in another PR, if time !